src/platform/lumin-runtime/elements/builders/image-builder.js

```diff
@@ -130,6 +141,41 @@ export class ImageBuilder extends UiNodeBuilder {
     }
   }
 
+  _createSpinner (prism) {
+    const spinner = this._createNode(ui.UiLoadingSpinner, 'Create', prism, ui.LoadingSpinnerType.k2dSpriteAnimation);
+    const [w, h] = this._callNodeFunction(spinner, 'getSize');
+    const [x, y, z] = this._callNodeFunction(spinner, 'getLocalPosition');
+
+    this._callNodeAction(spinner, 'setLocalPosition', [x - (w / 2), y - (h / 2), z]);
+    return spinner;
+  }
+
+  async _downloadResource (url, path, element, prism) {
+    // Set color mask
+    this._callNodeAction(element, 'setColor', [0.1, 0.1, 0.1, 0.1]);
+
+    // Add downloading spinner
+    const spinner = this._createSpinner(prism);
+    this._callNodeAction(element, 'addChild', spinner);
+
+    // Fetch the remote image
+    const filePath = await saveResource(url, path);
```

Comment:
we should try/catch around saveResource since this can throw exceptions for network issues. Eventually we should display a broken placeholder image on errors. For now we should at least log the error and remove the spinner to indicate download is no longer in progress.

Reply:
What happens if we the component is destroyed while we are downloading an image? For example, if we are paging through content in the UI and flip to the next page as it loads. Can we end up with invalid Kali node references?